fcrepo-auth-webac/src/test/java/org/fcrepo/integration/auth/webac/WebACRecipesIT.java

```diff
@@ -1325,4 +1332,596 @@ public void testCreateAclWithAccessToClassForBinary() throws Exception {
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(patchIndirect2));
+
+        final String patch_insert_relation = "prefix ldp: <http://www.w3.org/ns/ldp#> \n" +
+                "INSERT DATA { <> ldp:membershipResource <" + targetResource + "> }";
+        final HttpPatch patchIndirect3 = new HttpPatch(indirectUri);
+        setAuth(patchIndirect3, username);
+        patchIndirect3.setEntity(new StringEntity(patch_insert_relation, sparqlContentType));
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(patchIndirect3));
+
+        // Patch the indirect to the readonly target as admin
+        final HttpPatch patchAsAdmin = new HttpPatch(indirectUri);
+        setAuth(patchAsAdmin, "fedoraAdmin");
+        patchAsAdmin.setEntity(new StringEntity(patch_insert_relation, sparqlContentType));
+        assertEquals(HttpStatus.SC_NO_CONTENT, getStatus(patchAsAdmin));
+
+        // Try to POST a child as user
+        final HttpPost postChild = new HttpPost(indirectUri);
+        final String postTarget = "@prefix ldp: <http://www.w3.org/ns/ldp#> .\n" +
+                "@prefix test: <http://example.org/test#> .\n\n" +
+                "<> test:something <" + tempTarget + "> .";
+        final HttpEntity putPostChild = new StringEntity(postTarget, turtleContentType);
+        setAuth(postChild, username);
+        postChild.setEntity(putPostChild);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(postChild));
+
+        // Try to PUT a child as user
+        final String id = getRandomUniqueId();
+        final HttpPut putChild = new HttpPut(indirectUri + "/" + id);
+        setAuth(putChild, username);
+        putChild.setEntity(putPostChild);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(putChild));
+
+        // Put the child as Admin
+        setAuth(putChild, "fedoraAdmin");
+        assertEquals(HttpStatus.SC_CREATED, getStatus(putChild));
+
+        // Try to delete the child as user
+        final HttpDelete deleteChild = new HttpDelete(indirectUri + "/" + id);
+        setAuth(deleteChild, username);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(deleteChild));
+
+        // Try to delete the indirect container
+        final HttpDelete deleteIndirect = new HttpDelete(indirectUri);
+        setAuth(deleteIndirect, username);
+        assertEquals(HttpStatus.SC_FORBIDDEN, getStatus(deleteIndirect));
+
+        // Ensure we can still write to the writeable resource.
+        testCanWrite(writeableResource, username);
+
+    }
+
+    @Test
+    public void testIndirectRelationshipOK() throws IOException {
+        final String targetResource = "/rest/" + getRandomUniqueId();
+        final String writeableResource = "/rest/" + getRandomUniqueId();
+        final String username = "user28";
+
+        final String targetUri = ingestObj(targetResource);
+
+        final String readonlyString = "@prefix acl: <http://www.w3.org/ns/auth/acl#> .\n" +
```

Comment:
This variable name is a bit misleading ("readonlyString")... since the ACL actually specifies both `acl:Read` and `acl:Write`.
Suggestion to change the ACL to actually be read-only.

Reply:
Sorry I'll review this, I had written the `testIndirectRelationshipForbidden` test and then copied the code to make this test. Clearly missed some comments.